### PR TITLE
Background job to cleanup leftover chunked uploads

### PR DESCRIPTION
**Pull request by rullzer**
If a chunked uploads fails for some reason the chunks should be cleaned up at some point in time. Else they can quickly fillup a disk.

When a chunked collection is created a backgroundjob is inserted.
This job is deleted when the chunked upload completes sucsessfully

The job will check if all files have an mtime of more than 24 hours. If so it will kill the folder. So as long as you upload 1 chunk every 24 hours it will not kill them.

TODO:
- [x] Proper DI
- [x] Add an initial repair step that inserts a backgroundjob for all the existing upload folders
   * probably best to have a background job for all the users inserted that does this

CC: @MorrisJobke since we discussed this recently

### CODE REVIEW

File: apps/dav/lib/BackgroundJob/UploadCleanup.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+			/** @var Folder $uploadFolder */
+			$uploadFolder = $uploads->get($folder);
+		} catch (NotFoundException $e) {
+			$this->jobList->remove(self::class, $argument);
+			return;
+		}
+
+		$files = $uploadFolder->getDirectoryListing();
+
+		//Remove if all files have an mtime of more than a day
```
Comment (MorrisJobke) [inline apps/dav/lib/BackgroundJob/UploadCleanup.php]:
```suggestion
		// Remove if all files have an mtime of more than a day
```

File: apps/dav/lib/BackgroundJob/UploadCleanup.php [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+		$time = $this->time->getTime() - 60*60*24;
```
Comment (MorrisJobke) [inline apps/dav/lib/BackgroundJob/UploadCleanup.php]:
```suggestion
		$time = $this->time->getTime() - 60 * 60 * 24;
```

**Review by MorrisJobke — CHANGES_REQUESTED** [review]
Code makes sense so far 👍

**Comment by MorrisJobke** [comment]
I guess this is addressing #8759

**Comment by rullzer** [comment]
@MorrisJobke yes. But this is only for upload using the new chunked upload. However most are now days.

**Review by blizzz — APPROVED** [review]
makes sense

**Review by MorrisJobke — APPROVED** [review]
Code makes sense 👍

**Pull request merged**